mesonbuild/interpreterbase/interpreterbase.py

```diff
@@ -367,6 +370,10 @@ def evaluate_ternary(self, node: mparser.TernaryNode) -> T.Optional[InterpreterO
         else:
             return self.evaluate_statement(node.falseblock)
 
+    @FeatureNew('multiline format strings', '0.63.0')
+    def evaluate_multiline_fstring(self, node: mparser.MultilineFormatStringNode) -> InterpreterObject:
+        return self.evaluate_fstring(node)
```

Comment:
You can check in evaluate_fstring whether it's a multiline string and use `FeatureNew.single_use()` with a manually constructed message string.

This might save the need to add another AST token, but the question is whether that can be reliably detected. We can't have a literal newline character in a regular single single-quote string as written in meson.build, but is there any other way to end up with one and have it tokenized as a FormatStringNode?

e.g. if you format the newline escape sequence into a string, that should just result in a StringNode, not a FormatStringNode. To differentiate between a string and a multiline string we would need to know whether the newline char came from a literal or via string building and formatting, but an f-string doesn't have that concern... I think.

Reply:
I'm inclined to leave it with the extra AST node so that the code to detect a multline format string is in one location (the parser). Happy to change though if you feel otherwise.

Reply:
I'm not really picky, I guess. It would probably be easier to implement without the extra token type, but on the other hand it's already implemented...